Calculator.Logic/CalculatorLogic.cs

```diff
@@ -0,0 +1,302 @@
+        {
+            var termOne = CalculatorConverterStringToNumber.ConvertToUInt(firstTerm);
+            if (termOne % 2 == 0)
+            {
+                result.Message = "The number is even.";
+                result.Value = $"{termOne}";
+            }
+            else
+            {
+                result.Message = "The number is odd.";
+                result.Value = $"{termOne}";
+            }
+        }
+        private static void ExecuteGetMirroredOperation(string firstTerm, CalculatorResult result)
+        {
+            var termOne = CalculatorConverterStringToNumber.ConvertToDouble(firstTerm);
+            if (IsNaturalNumber(termOne))
+            {
+                uint number = Convert.ToUInt32(firstTerm);
+                var resultValue = GetMirroredNumber(number);
+                result.Value = resultValue.ToString();
+            }
+            else
+            {
+                result.Message = "Please enter a natural number.";
+            }
+        }
+        private static uint GetMirroredNumber(uint number)
+        {
+            uint mirrored = 0;
+            while (number != 0)
+            {
+                mirrored = mirrored * 10 + number % 10;
+                number /= 10;
+            }
+            return mirrored;
+        }
+        private static void ExecutePrimeOperation(string firstTerm, CalculatorResult result)
+        {
+            var termOne = CalculatorConverterStringToNumber.ConvertToDouble(firstTerm);
+            if (IsNaturalNumber(termOne))
+            {
+                uint convertedFirstTerm = Convert.ToUInt32(firstTerm);
+                if (convertedFirstTerm <= 1)
+                {
+                    result.Message = $"{convertedFirstTerm} is not a prime number.";
+                    result.Value = $"{convertedFirstTerm}";
+                }
+                else if (convertedFirstTerm == 2)
+                {
+                    result.Message = $"{convertedFirstTerm} is  a prime number.";
+                    result.Value = $"{convertedFirstTerm}";
+
+                }
+                else
+                {
+                    if (HasDivisors(convertedFirstTerm))
+                    {
+                        result.Message = $"{convertedFirstTerm} is  a prime number.";
+                        result.Value = $"{convertedFirstTerm}";
+
+                    }
+                    else
+                    {
+                        result.Message = $"{convertedFirstTerm} is not a prime number.";
+                        result.Value = $"{convertedFirstTerm}";
+
+                    }
+                }
+            }
+            else
+            {
+                result.Message = $" {firstTerm} is not a natural number.";
+                result.Value = $"{firstTerm}";
+
+            }
+        }
+        private static void ExecutePalindromeOrSuperPalindromeOperation(string firstTerm, CalculatorResult result)
```

Comment:
for each of this execute methods, instead of seding the result as a parameter and the method returning void, change it to return a calculatorResult, since that's what you need. Do not put as parameters what you actually want to return

Reply:
done